src/randgraphs.jl

```diff
@@ -193,6 +193,130 @@ function barabasi_albert(n::Integer, k::Integer; is_directed=false, seed::Int=-1
+    n = length(fitness)
+    m == 0 && return Graph(n)
+    nodes = 0
+    for f in fitness
+        # sanity check for the fitness
+        f < zero(T) && error("fitness scores must be non-negative")
+        f > zero(T) && (nodes += 1)
+    end
+    # avoid getting into an infinite loop when too many edges are requested
+    max_no_of_edges = div(nodes*(nodes-1), 2)
+    @assert(m <= max_no_of_edges, "too many edges requested")
+    # calculate the cumulative fitness scores
+    cum_fitness = cumsum(fitness)
+    g = Graph(n)
+    _create_static_fitness_graph!(g, m, cum_fitness, cum_fitness, seed)
+    return g
+end
+
+function static_fitness_model{T<:Real,S<:Real}(m::Int, fitness_out::Vector{T}, fitness_in::Vector{S}; seed::Int=-1)
+    @assert(m >= 0, "invalid number of edges")
+    n = length(fitness_out)
+    @assert(length(fitness_in) == n, "fitness_in must have the same size as fitness_out")
+    m == 0 && return DiGraph(n)
+    # avoid getting into an infinite loop when too many edges are requested
+    outnodes = innodes = nodes = 0
+    @inbounds for i=1:n
+        # sanity check for the fitness
+        (fitness_out[i] < zero(T) || fitness_in[i] < zero(S)) && error("fitness scores must be non-negative")
+        fitness_out[i] > zero(T) && (outnodes += 1)
+        fitness_in[i] > zero(S) && (innodes += 1)
+        (fitness_out[i] > zero(T) && fitness_in[i] > zero(S)) && (nodes += 1)
+    end
+    max_no_of_edges = outnodes*innodes - nodes
+    @assert(m <= max_no_of_edges, "too many edges requested")
+    # calculate the cumulative fitness scores
+    cum_fitness_out = cumsum(fitness_out)
+    cum_fitness_in = cumsum(fitness_in)
+    g = DiGraph(n)
+    _create_static_fitness_graph!(g, m, cum_fitness_out, cum_fitness_in, seed)
+    return g
+end
+
+function _create_static_fitness_graph!{T<:Real,S<:Real}(g::SimpleGraph, m::Int, cum_fitness_out::Vector{T}, cum_fitness_in::Vector{S}, seed::Int)
+    rng = getRNG(seed)
+    max_out = cum_fitness_out[end]
+    max_in = cum_fitness_in[end]
+    while m > 0
+        source = searchsortedfirst(cum_fitness_out, rand(rng)*max_out)
+        target = searchsortedfirst(cum_fitness_in, rand(rng)*max_in)
+        # skip if loop edge
+        (source == target) && continue
+        edge = Edge(source, target)
+        # is there already an edge? If so, try again
+        has_edge(g, edge) && continue
+        add_edge!(g, edge)
+        m -= 1
+    end
+end
+
+"""
+    function static_scale_free(n::Int, m::Int, alpha::Float64; seed::Int=-1, finite_size_correction::Bool=false)
+
+Generates a non-growing random graph with expected power-law degree distributions.
+`n` is the number of nodes in the generated graph. `m` is the number of edges in the generated graph.
+`alpha` is the power law exponent of digree distribution.
+`finite_size_correction` determines whether to use the proposed finite size correction of Cho et al. Time complexity is O(|V| + |E| log |E|).
```

Comment:
->
Generates a random graph with `n` vertices, `m` edges and expected power-law degree distribution with exponent `alpha`. `finite_size_correction` determines whether to use the finite size correction proposed by Cho et al. 
This generator calls internally the `static_fitness_model` function. Time complexity is O(|V| + |E| log |E|).